### PR TITLE
Github Actions uses fixed job names for branch protection rules

### PR DESCRIPTION
**Pull request by adamrtalbot**
Change Github Actions jobs to use fixed names for branch protection rules. The matrix parameter will still be in the panel as:

jobname (tag, profile)

So there is some visibility, but now the branch protection rules _will actually work_.

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: .github/workflows/nf-core-linting.yml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 name: nf-core linting
+run-name: nf-core linting
```
Comment (drpatelh) [inline .github/workflows/nf-core-linting.yml:2]:
```suggestion
run-name: nf-core lint
```

File: .github/workflows/nf-core-linting.yml [hunk 2]
```diff
@@ -14,7 +15,7 @@ concurrency:
 
 jobs:
   changes:
-    name: Check for changes
+    name: check-for-changes
```
Comment (drpatelh) [inline .github/workflows/nf-core-linting.yml:18]:
```suggestion
    name: check-changes
```

File: .github/workflows/nf-test.yml [hunk 3]
```diff
@@ -15,7 +16,7 @@ concurrency:
 
 jobs:
   changes:
-    name: Check for changes
+    name: check-for-changes
```
Comment (drpatelh) [inline .github/workflows/nf-test.yml:19]:
```suggestion
    name: check-changes
```

File: .github/workflows/nf-test.yml [hunk 4]
```diff
@@ -28,10 +29,9 @@ jobs:
         with:
           filters: "tests/config/nftest_modules.yml"
 
-  test:
+  nf_test:
```
Comment (drpatelh) [inline .github/workflows/nf-test.yml:32]:
```suggestion
  nf-test:
```

File: .github/workflows/pytest-workflow.yml [hunk 5]
```diff
@@ -15,7 +16,7 @@ concurrency:
 
 jobs:
   changes:
-    name: Check for changes
+    name: check-for-changes
```
Comment (drpatelh) [inline .github/workflows/pytest-workflow.yml:19]:
```suggestion
    name: check-changes
```

File: .github/workflows/pytest-workflow.yml [hunk 6]
```diff
@@ -28,10 +29,9 @@ jobs:
         with:
           filters: "tests/config/pytest_modules.yml"
 
-  test:
+  pytest_workflow:
```
Comment (drpatelh) [inline .github/workflows/pytest-workflow.yml:32]:
```suggestion
  pytest:
```

File: .github/workflows/pytest-workflow.yml [hunk 7]
```diff
@@ -28,10 +29,9 @@ jobs:
...
     runs-on: ubuntu-20.04
-
-    name: ${{ matrix.tags }} ${{ matrix.profile }}
+    name: pytest-workflow
```
Comment (drpatelh) [inline .github/workflows/pytest-workflow.yml:34]:
```suggestion
    name: pytest
```

File: .github/workflows/nf-core-linting.yml [hunk 8]
```diff
@@ -1,4 +1,5 @@
 name: nf-core linting
```
Comment (drpatelh) [inline .github/workflows/nf-core-linting.yml:1]:
```suggestion
name: nf-core lint
```

File: .github/workflows/code-linting.yml [hunk 9]
```diff
@@ -1,4 +1,4 @@
-name: Code Linting
+name: code-linting
```
Comment (drpatelh) [inline .github/workflows/code-linting.yml:1]:
```suggestion
name: code-lint
```

Comment (drpatelh) [reply .github/workflows/code-linting.yml:1]: For consistency

**Comment by drpatelh** [comment]
Tricky getting it right until you see the actual check names that are triggered.

**Review by drpatelh — APPROVED** [review]
LGTM! 🕺🏽 Thank you!

**Review by mirpedrol — APPROVED** [review]
LGTM :)

**Pull request merged**